### PR TITLE
refactor zcl_excel_writer_2007 - create_xl_sheet method

### PR DESCRIPTION
**Pull request by Dwivedirahul44**
issue #915.
massive refactor, tool-assisted and manual.
I have tried my best to avoid adding side effects, and no apparent issue found so far. 
please review carefully and provide feedback.

### CODE REVIEW

File: src/zcl_excel_writer_2007.clas.abap [hunk 1]
```diff
@@ -3674,275 +3720,26 @@ CLASS zcl_excel_writer_2007 IMPLEMENTATION.
 
 
   METHOD create_xl_sheet.
```
Comment (Dwivedirahul44) [inline src/zcl_excel_writer_2007.clas.abap:3722]: Focal point of the refactor. Subnode creation delegated to separate method

File: src/zcl_excel_writer_2007.clas.abap [hunk 2]
```diff
@@ -3987,4258 +3808,4593 @@ CLASS zcl_excel_writer_2007 IMPLEMENTATION.
...
+                                                         parent = io_document ).
       lo_element_2->set_attribute_ns( name  = lc_xml_attr_fittopage
-                                    value = `1` ).
+                                      value = `1` ).
       lo_element->append_child( new_child = lo_element_2 ). " pageSetupPr node
     ENDIF.
 
     lo_element_root->append_child( new_child = lo_element ).
+  ENDMETHOD.
+  METHOD create_xl_sheet_subnodes.
```
Comment (Dwivedirahul44) [inline src/zcl_excel_writer_2007.clas.abap:3820]: New method which got broken further based on individual subnodes.

File: src/zcl_excel_writer_2007.clas.abap [hunk 3]
```diff
@@ -261,6 +260,65 @@ CLASS zcl_excel_writer_2007 DEFINITION
         !ip_flag          TYPE flag
       RETURNING
         VALUE(ep_boolean) TYPE tv_charbool  .
+    METHODS:
+      create_xl_sheet_subnodes       IMPORTING io_worksheet TYPE REF TO zcl_excel_worksheet
```
Comment (Dwivedirahul44) [inline src/zcl_excel_writer_2007.clas.abap:264]: newly introduced methods

**Review by Dwivedirahul44 — COMMENTED** [review]
Main change points : 
[create_xl_sheet logic delegation](https://github.com/abap2xlsx/abap2xlsx/pull/1068/files#r1002271687)
[New method capturing the delegated logic](https://github.com/abap2xlsx/abap2xlsx/pull/1068/files#r1002272812)
[newly introduced methods](https://github.com/abap2xlsx/abap2xlsx/pull/1068/files#r1002276736)

**Comment by sandraros** [comment]
Thanks. As side remark, I see that you pretty-printed (manually I guess) a lot of lines, the diff is huge, I can't focus on the essential part. So, I can't be sure that you didn't break something, next time it would be best to do it as a separate PR just about pretty print.

**Comment by sandraros** [comment]
Another issue with the GitHub diff is that it's completely messed up, all new methods appear as a modification of unchanged methods. Is it possible to improve the changes so that the diff shows what has really changed? If needed, intermediate PRs could be created and merged.

**Comment by Dwivedirahul44** [comment]
I agree it's a mess to review, I had to manually extract individual subnodes code to individual methods as extract method tool was going haywire. 

Should I retract this pr and submit with minimal changes and build upon it review after review? Thag should make your life easier reviewing the changes.

**Comment by Dwivedirahul44** [comment]
It will be slow but stable process.
I will add new pr if you agree to it.
Planned time-line -
Pr1 : add wrapper method for subnodes creation encapsulating all the code and it's call in create_xl_sheet method.
Review and merge pr.
Pr2 : breaks wrapper method logic into individual methods handling each sub nodes.
Review and merge.
Pr3 : delete unused variables
Review and merge
Pr4 : pretty print
Review and merge.

**Comment by sandraros** [comment]
Maybe 2 PRs could be sufficient:
- one for Pr1+Pr2 (reuse current PR 1068). Maybe creating a new class would help in making the diff better.
- one for Pr3+Pr4.

**Comment by Dwivedirahul44** [comment]
> 

alright, I didn't understand the part about creating a new class, do you want me to make a copy and refactor it there? and leave the original intact? I tried extracting the whole logic to the local helper class but couldn't decouple he logic enough to do so.

**Comment by sandraros** [comment]
Not a copy. A local helper class would be fine. If needed, you might make it a friend (there are many anti-patterns already in abap2xlsx, I don't think it's a big deal to have another one).

**Comment by Dwivedirahul44** [comment]
this PR has gone messy , its easier for me to close this and a fresh PR with changes.

**Pull request closed without merging**